### PR TITLE
Adding ellipse tank unit --Task 2

### PR DESCRIPTION
**Pull request by XiaoxiangRdM**
新添加一个椭圆坦克单元，加了三个技能
前两个是向前/向后瞬移并发射一排八个子弹，冷却时间短

后一个是生命值恢复，冷却时间长且冷却时要求其他技能冷却结束

### CODE REVIEW

File: src/battle_game/core/selectable_units.cpp [hunk 1]
```diff
@@ -39,6 +39,11 @@ void GameCore::GeneratePrimaryUnitList() {
   ADD_SELECTABLE_UNIT(unit::WhaoooooTank);
   ADD_SELECTABLE_UNIT(unit::SquareTank);
 
+
+
+  
+  ADD_SELECTABLE_UNIT(unit::XXRdMTank);
+
```
Comment (LazyJazz) [inline src/battle_game/core/selectable_units.cpp:46]: remove unnecessary lines.

File: src/battle_game/core/units/units.h [hunk 2]
```diff
@@ -8,3 +8,8 @@
 #include "battle_game/core/units/three_body_man.h"
 #include "battle_game/core/units/tiny_tank.h"
 #include "battle_game/core/units/triple_shot_tank.h"
+
+
+
+
```
Comment (LazyJazz) [inline src/battle_game/core/units/units.h:14]: These too

File: src/battle_game/core/units/XiaoxiangRdM_tank.h [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  [[nodiscard]] const char *UnitName() const override;
+  [[nodiscard]] const char *Author() const override;
+
+  float turret_rotation_{0.0f};
+  uint32_t fire_count_down_{0};
+  uint32_t teleport_count_down_{360};
+  uint32_t healthcare_count_down_{1200};
+
+  float x_0 = 0.8; //semi-minor axis
+  float y_0 = 1.2; //semi-major axis
```
Comment (LazyJazz) [inline src/battle_game/core/units/XiaoxiangRdM_tank.h:32]: lower_case_ ending with an underscore

File: src/battle_game/core/units/XiaoxiangRdM_tank.cpp [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+      if (input_data.key_down[GLFW_KEY_E]) TeleportFClick();
+      else if(input_data.key_down[GLFW_KEY_Q]) TeleportBClick();
+    }
+  }
+}
+
+void XXRdMTank::HealthcareClick(){
+  teleport_count_down_ = 3 * kTickPerSecond;
+  healthcare_count_down_ = 20 * kTickPerSecond;
+  GameCore x;
```
Comment (LazyJazz) [inline src/battle_game/core/units/XiaoxiangRdM_tank.cpp:235]: What are you doing?

**Review by LazyJazz — CHANGES_REQUESTED** [review]
Fix the issues.

**Comment by XiaoxiangRdM** [comment]
issues are fixed

File: src/battle_game/core/units/XiaoxiangRdM_tank.cpp [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+  if (player) {
+    auto &input_data = player->GetInputData();
+    auto diff = input_data.mouse_cursor_position - position_;
+    if (glm::length(diff) < 1e-4) {
+      turret_rotation_ = rotation_;
+    } else {
+      turret_rotation_ = std::atan2(diff.y, diff.x) - glm::radians(90.0f);
+    }
+  }
+}
```
Comment (LazyJazz) [inline src/battle_game/core/units/XiaoxiangRdM_tank.cpp:166]: inherit from unit::Tank if you are reusing these functions.

**Review by LazyJazz — CHANGES_REQUESTED** [review]

File: src/battle_game/core/units/XiaoxiangRdM_tank.cpp (hunk 5 above)
Comment (XiaoxiangRdM) [reply src/battle_game/core/units/XiaoxiangRdM_tank.cpp]: fixed as required

**Review by LazyJazz — APPROVED** [review]
LGTM!

**Pull request closed without merging**